frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts

```diff
@@ -383,15 +383,29 @@ type KeyValuePair = {
   key: string;
   value: string;
 };
-export const getSecretAnnotations = (annotation: KeyValuePair) => {
-  const annotations = {};
+
+const getAnnotationKey = (secretType: string, suffix: number) => {
   const annotationPrefix = 'tekton.dev';
-  if (annotation?.key === SecretAnnotationId.Git) {
-    annotations[`${annotationPrefix}/${SecretAnnotationId.Git}-0`] = annotation?.value;
-  } else if (annotation?.key === SecretAnnotationId.Image) {
-    annotations[`${annotationPrefix}/${SecretAnnotationId.Image}-0`] = annotation?.value;
+  if (secretType === SecretAnnotationId.Git) {
+    return `${annotationPrefix}/${SecretAnnotationId.Git}-${suffix}`;
+  }
+  if (secretType === SecretAnnotationId.Image) {
+    return `${annotationPrefix}/${SecretAnnotationId.Image}-${suffix}`;
   }
-  return annotations;
+  return null;
+};
+
+export const getSecretAnnotations = (
+  annotation: KeyValuePair,
+  existingAnnotations: { [key: string]: string } = {},
+) => {
+  let count = 0;
+  let annotationKey = getAnnotationKey(annotation?.key, count);
+  while (existingAnnotations[annotationKey]) {
+    annotationKey = getAnnotationKey(annotation?.key, ++count);
+  }
+
+  return { ...existingAnnotations, [annotationKey]: annotation?.value };
```

Comment:
do we need to maintain count or can use a random number to generate a new unique key and `getAnnotationKey` even return null as key if  `annotation?.key` is not a match. Does below make sense can avoid function `getAnnotationKey`  as well

```suggestion
export const getSecretAnnotations = (
  annotation: KeyValuePair,
  existingAnnotations: { [key: string]: string } = {},
) => {
 if(!annotation?.key || ![SecretAnnotationId.Git, SecretAnnotationId.Image].includes(annotation?.key )) return existingAnnotations;
  const annotationPrefix = 'tekton.dev';
  const suffix = Math.round(Math.random() * 10000);
  const annotationKey = `tekton.dev/${secretType}-${suffix}`;
  return { ...existingAnnotations, [annotationKey]: annotation?.value };
}
```

Reply:
In Tekton documentation, says the credential annotation it must begin with  `tekton.dev/git-` or `tekton.dev/docker-`,  technically I guess we can add any random number, however adding the count keeps us consistent with how they outlined the usage in docs and the yaml users might get confused on seeing random number at the suffix. I would say lets go with count.

https://github.com/tektoncd/pipeline/blob/main/docs/auth.md#understanding-credential-selection

Reply:
makes sense, let's handle the null scenario for this `getAnnotationKey` and add / update specs for utils  cc @rottencandy

Reply:
Thanks. Updated to handle invalid key scenario.